app/models/post.rb

```diff
@@ -1,2 +1,21 @@
 class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User'
+
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_counter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  after_create :update_posts_counter
+  after_destroy :update_posts_counter
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] [OPTIONAL] It's a brilliant one calling the `after_save` Rail Active Callback to update posts every time a new post has been saved but it would be highly impressionable if you make the `update_posts_counter` method `private`.
- Just so you don't forget, remember to make `#five_most_recent_comments` remain `public` for accessibility.

Reply:
I have also implemented this